src/App.tsx

```diff
@@ -1,14 +1,47 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { SortType } from './enums/SortType';
+import { filterTodos } from './helpers/filterTodos';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentTodoId, setCurrentTodoId] = useState(0);
+  const [listLoad, setListLoad] = useState(false);
+  const [query, setQuery] = useState('');
+  const [sortType, setSortType] = useState(SortType.All);
+
+  async function getTodosFromServer() {
+    setListLoad(true);
+    try {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+    } catch (error) {
+      setTodos([]);
+    }
+
+    setListLoad(false);
+  }
+
+  useEffect(() => {
+    getTodosFromServer();
+  }, []);
+
+  const currentTodo = todos.find(todo => todo.id === currentTodoId);
+
```

Comment:
You can pass the entire selected todo instead of just the id, and then avoid searching for the todo by id. If you still want to keep the search function, you can move it to the helper functions.